File: webui/src/main/scala/com/taskadapter/webui/pages/ModalWindow.scala
package com.taskadapter.webui.pages

import com.vaadin.flow.component.Component
import com.vaadin.flow.component.dialog.Dialog

object ModalWindow {

  def showDialog(component: Component): Dialog = {
    val dialog = new Dialog()
    dialog.setModal(true)
    dialog.setCloseOnEsc(true)
    dialog.setCloseOnOutsideClick(true)
    dialog.add(component)
    dialog.open()
    dialog
  }
}

File: webui/src/main/java/com/taskadapter/webui/pages/ModalWindow.java
package com.taskadapter.webui.pages;

import com.vaadin.flow.component.Component;
import com.vaadin.flow.component.dialog.Dialog;

public class ModalWindow {
    public static Dialog showDialog(Component component) {
        var dialog = new Dialog();
        dialog.setModal(true);
        dialog.setCloseOnEsc(true);
        dialog.setCloseOnOutsideClick(true);
        dialog.add(component);
        dialog.open();
        return dialog;
    }
}
